tests/zfs-tests/tests/functional/cli_root/zdb/zdb_args_neg.ksh

```diff
@@ -57,7 +57,7 @@ set -A args "create" "add" "destroy" "import fakepool" \
     "add raidz1 fakepool" "add raidz2 fakepool" \
     "setvprop" "blah blah" "-%" "--?" "-*" "-=" \
     "-a" "-f" "-g" "-j" "-n" "-o" "-p" "-p /tmp" \
-    "-t" "-w" "-z" "-E" "-H" "-I" "-J" "-K" \
+    "-t" "-w" "-z" "-E" "-H" "-I" "-J" \
     "-Q" "-R" "-T" "-W"
 
 log_assert "Execute zdb using invalid parameters."
```

Comment:
Let's also add a new `tests/functional/zdb_encrypted.ksh` test case which exercises the command line option and verifies basic functionality.  There are several existing test like `tests/functional/cli_root/zfs_mount/zfs_mount_encrypted.ksh` you can use as a starting point.

Reply:
Done in c66ecbb! We'll see if it passes :sweat_smile:

Reply:
It looks like it worked.  Then I think all that's left is to sort out these few build failures, http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20%28TEST%29/builds/23556/steps/shell_1/logs/make

Reply:
Most of these are:

```
cmd/zdb/zdb.c: In function 'sublivelist_verify_blkptr':
cmd/zdb/zdb.c:297:4: error: 'for' loop initial declarations are only allowed in C99 mode
    for (int i = 0; i < SPA_DVAS_PER_BP; i++) {
    ^
```

There's only one that's possibly mine:

```
cmd/zdb/zdb.c: In function 'zdb_derive_key':
cmd/zdb/zdb.c:3049:4: warning: format '%x' expects argument of type 'unsigned int *', but argument 3 has type 'unsigned char *' [-Wformat=]
    if (sscanf(&key_material[i], "%02x", &c) != 1)
    ^
```

But even then, I'm copying the same piece of code from [`libzfs_crypto.c`](https://github.com/openzfs/zfs/blob/master/lib/libzfs/libzfs_crypto.c#L162).

I wondered if including `openssl/evp.h` was changing compiler options but `libzfs_crypto.c` also includes that. It shouldn't be a wider build change; I'm adding `LIBCRYPTO_CFLAGS` to zdb but `config.log` says its empty.

So I'm not sure where this is coming from, or why its coming up now. I'm believe you, but you might have to point me at it.

Reply:
Interestingly, FreeBSD builder is failing in a similar shape: http://build.zfsonlinux.org/builders/FreeBSD%20main%20amd64%20%28TEST%29/builds/5668/steps/shell_1/logs/make

Not exactly the same, but C99 warnings in zdb.c.

I wonder if zdb is hooked up to the build in a slightly different way and we're now bumping into something. I'll do a build on FreeBSD at least, see what I can see.

Reply:
Yeah, fine there too, and on FreeBSD 13 clang is a couple of versions behind the one in the buildbot (unlike the CentOS builder, which was GCC 4.5).

So so far this all says "buildbot is weird" but I can't yet imagine a kind of weirdness that gets us here. I'll think on it more today.

(if make was run with `V=1` the compiler flags in use might reveal more?)

Reply:
This appears to be an issue with automake-1.13.4 which is what's used on CentOS 7.  Setting `zdb_CFLAGS` overwrites all of the default `AM_FLAGS` which means `std=gnu99` no longer gets passed.  The same thing can happen with `zdb_CPPFLAGS` so it always includes `AM_CPPFLAGS`.  This small change resolves the issue for me.

```diff
--- a/cmd/zdb/Makefile.am
+++ b/cmd/zdb/Makefile.am
@@ -1,5 +1,5 @@
 zdb_CPPFLAGS = $(AM_CPPFLAGS) $(FORCEDEBUG_CPPFLAGS)
-zdb_CFLAGS   = $(LIBCRYPTO_CFLAGS)
+zdb_CFLAGS   = $(AM_CFLAGS) $(LIBCRYPTO_CFLAGS)
 
 sbin_PROGRAMS   += zdb
 CPPCHECKTARGETS += zdb
```

Reply:
Interesting, good find. Seems like `ztest_CFLAGS` does this too. Fix pushed, lets see how the test run goes. Cheers!